keras_cv/metrics/f_scores.py

```diff
@@ -0,0 +1,356 @@
+        self.false_negatives = _zero_wt_init("false_negatives")
+        self.weights_intermediate = _zero_wt_init("weights_intermediate")
+
+    def reset_state(self):
+        self.true_positives.assign(tf.zeros_like(self.true_positives))
+        self.false_positives.assign(tf.zeros_like(self.false_positives))
+        self.false_negatives.assign(tf.zeros_like(self.false_negatives))
+        self.weights_intermediate.assign(tf.zeros_like(self.weights_intermediate))
+
+    @tf.function
+    def update_state(self, y_true, y_pred, sample_weight=None):
+        if self.threshold is None:
+            threshold = tf.reduce_max(y_pred, axis=-1, keepdims=True)
+            # make sure [0, 0, 0] doesn't become [1, 1, 1]
+            # Use abs(x) > eps, instead of x != 0 to check for zero
+            y_pred = tf.logical_and(y_pred >= threshold, tf.abs(y_pred) > 1e-12)
+        else:
+            y_pred = y_pred > self.threshold
+
+        y_true = tf.cast(y_true, self.dtype)
+        y_pred = tf.cast(y_pred, self.dtype)
+
+        def _weighted_sum(val, sample_weight):
+            if sample_weight is not None:
+                val = tf.math.multiply(val, tf.expand_dims(sample_weight, 1))
+            return tf.reduce_sum(val, axis=self.axis)
+
+        self.true_positives.assign_add(_weighted_sum(y_pred * y_true, sample_weight))
+        self.false_positives.assign_add(
+            _weighted_sum(y_pred * (1 - y_true), sample_weight)
+        )
+        self.false_negatives.assign_add(
+            _weighted_sum((1 - y_pred) * y_true, sample_weight)
+        )
+        self.weights_intermediate.assign_add(_weighted_sum(y_true, sample_weight))
+
+    @tf.function
+    def result(self):
+        precision = tf.math.divide_no_nan(
+            self.true_positives, self.true_positives + self.false_positives
+        )
+        recall = tf.math.divide_no_nan(
+            self.true_positives, self.true_positives + self.false_negatives
+        )
+
+        mul_value = precision * recall
+        add_value = (tf.math.square(self.beta) * precision) + recall
+        mean = tf.math.divide_no_nan(mul_value, add_value)
+        f1_score = mean * (1 + tf.math.square(self.beta))
+
+        if self.average == "weighted":
+            weights = tf.math.divide_no_nan(
+                self.weights_intermediate, tf.reduce_sum(self.weights_intermediate)
+            )
+            f1_score = tf.reduce_sum(f1_score * weights)
+
+        elif self.average is not None:  # [micro, macro]
+            f1_score = tf.reduce_mean(f1_score)
+
+        return f1_score
+
+    def get_config(self):
+        """Returns the serializable config of the metric."""
+
+        config = {
+            "num_classes": self.num_classes,
+            "average": self.average,
+            "beta": self.beta,
+            "threshold": self.threshold,
+        }
+
+        base_config = super().get_config()
+        return {**base_config, **config}
+
+
+class F1Score(FBetaScore):
```

Comment:
lets use one file per metric.  it is easier to stay organized and read/review code in that way.  feel free to make an `f_scores` subdirectory